### DISCUSSION TITLE
Proposed Updates to HaC Mini Project

### DISCUSSION THREAD

**Discussion by Malumen** [osy86/HaC-Mini#568] [Ideas]
Hi @osy86 (or @osy ) and @goodbest !

Everything about this project is amazing. We all love and appreciate the work you guys do!

**I wanted to open up a discussion spot for potential features/changes to the HaC mini project**

1. `Itlwm.kext` falls short when compared to the proper implementation of `AirPortItlwm.kext`. The `Itlwm.kext` provides basic WiFi connectivity but also requires users to download `HeliPort.app`, which may not be possible since some users may not have ethernet access to use a wired connection in order to download and install `HeliPort.app` in the first place... In addition: users report `AirPortItlwm.kext` providing some features like `AirDrop` are working as well! The caveat is that `AirPortItlwm.kext`is OS-specific and requires enabling `SecureBootModel -> Default`. (on my own system, enabling SecureBootModel causes macOS to prompt for reinstall after reboot... this may be because you are not using ApECID = _0_)

-  Installer Pkg would have a checker for IntelWifi for High Sierra, Mojave, Catalina, or Big Sur, or could auto-check what OS is installed and pull the corresponding `AirPortItlwm.kext`.
-  New versions of `AirPortItlwm.kext 1.3.0 Stable` may provide wireless `ac` support or even `ax`.

2. New OpenCore 0.6.6+ offer a feature to replace `BootProtect` with `LauncherOption` and fully protects the boot process from being messed up by Windows or other OS updates. This is ideal for dual-booting scenarios.
3. Upcoming fan control features implemented when complete from SMC kexts.

**Comment by osy** [comment]
According to https://openintelwireless.github.io/itlwm/FAQ.html it seems like `AirPortItlwm.kext` is far less stable than `itlwm.kext`.

**Reply by Malumen** [reply]
> According to https://openintelwireless.github.io/itlwm/FAQ.html it seems like `AirPortItlwm.kext` is far less stable than `itlwm.kext`.

That page has not been updated in a long time. V1.3.0 is very stable and I have been using it daily on 3 systems (including v1.2.0 stable).
